jetty-openid/src/main/java/org/eclipse/jetty/security/openid/JwtDecoder.java

```diff
@@ -63,7 +66,10 @@ and the Token Endpoint (which it is in this flow), the TLS server validation
         if (LOG.isDebugEnabled())
             LOG.debug("JWT signature not validated {}", jwtSignature);
 
-        return (Map)JSON.parse(jwtClaimString);
+        Object parsedClaims = JSON.parse(jwtClaimString);
+        if (!(parsedClaims instanceof Map))
+            throw new IllegalStateException("Could not decode JSON for JWT claims.");
+        return (Map)parsedClaims;
```

Comment:
This method returns a typed map. These types should be checked as well before returning.